src/main/java/com/example/moviemanager/service/MovieServiceImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.example.moviemanager.service;
+
+import com.example.moviemanager.entity.Movie;
+import com.example.moviemanager.repo.MovieRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+    private final MovieRepository movieRepository;
+
+    @Autowired
+    public MovieServiceImpl(MovieRepository movieRepository) {
+        this.movieRepository = movieRepository;
+    }
+
+    @Override
+    public List<Movie> findAll() {
+        return movieRepository.findAll();
+    }
+
+    @Override
+    public Movie findById(int id) {
+        Optional<Movie> movie = movieRepository.findById(id);
+        if (movie.isEmpty()) {
+            throw new MovieNotFoundException("Could not find Movie by id - " + id);
+        }
+
+        return movie.get();
+    }
+
+    @Override
+    public void deleteById(int id) {
+        try {
+            movieRepository.deleteById(id);
+        } catch (EmptyResultDataAccessException e) {
+            throw new MovieNotFoundException("Could not find Movie by id - " + id);
```

Comment:
I think message should be more specific that you can't delete movie, because now it would not be clear if you tried to find movie by id or delete it by id